docs/source/federation-spec.mdx

```diff
@@ -0,0 +1,625 @@
+  upc: String!
+  name: String!
+}
+```
+
+
+```graphql title="Reviews subgraph"
+type Product @key(fields: "upc") {
+  upc: String!
+  reviews: [Review!]!
+}
+
+type Review {
+  score: Int!
+  description: String!
+}
+```
+
+</CodeColumns>
+
+If a subgraph defines any entities, it must also provide the graph router direct access to the fields of those entities. This requires the following:
+
+- Defining the `_Entity` union type, which must include all of the subgraph's defined entity types
+- Defining the `Query._entities` field and resolving it correctly
+
+Both of these definitions are part of the [subgraph schema specification](#subgraph-schema-specification).
+
+### Defining the `_Entity` union
+
+The `_Entity` union type is the only schema definition from the [subgraph schema specification](#subgraph-schema-specification) that a subgraph must generate _dynamically_ based on the schema it's provided. All other definitions are static and can be added exactly as shown.
+
+The `_Entity` union must include all of the entity types that are defined in the subgraph schema (and no _other_ types).
+
+> If a subgraph doesn't define _any_ entity types, then it should _not_ define the `_Entity` union.
+
+For example, consider this subgraph schema:
+
+```graphql title="schema.graphql"
+type Review @key(fields: "id") {
+  id: ID!
+  body: String
+  author: User
+  product: Product
+}
+
+type User @key(fields: "email") {
+  email: String!
+}
+
+type Product @key(fields: "upc") {
+  upc: String!
+}
+```
+
+All three of the types in this subgraph schema are entities (note their `@key` directives). Therefore, the subgraph service should add the following `_Entity` union definition to the schema:
+
+```graphql
+union _Entity = Review | User | Product
+```
+
+The `_Entity` union is used by the `Query._entities` field, which is covered next.
+
+### Understanding `Query._entities`
+
+> This section describes _how_ a graph router uses the `Query._entities` field to combine entity data from multiple subgraphs. If you aren't interested in the details, you can skip directly to [Resolving `Query._entities`](#resolving-query_entities).
+
+Any subgraph that defines at least one entity must automatically define and correctly resolve the `Query._entities` field:
+
+```graphql {2}
+type Query {
+  _entities(representations: [_Any!]!): [_Entity]!
+}
+```
+
+> If a subgraph doesn't define any entity types, then it should _not_ define the `Query._entities` field.
+
+The graph router uses this entry point to directly fetch fields of entity objects. It combines those fields with _other_ fields of the same entity that are returned by _other_ subgraphs.
+
+The `Query._entities` field takes a required `representations` argument, which is a list of entity **representations**. A representation is an object that contains all of an entity's `@key` fields, plus its `__typename` field. These are the fields that a subgraph can use to uniquely identify a particular instance of an entity.
```

Comment:
nit: entity can specify multiple `@key`s (it is a repeatable directive) so it will contain one of field sets that identity the target object